confgenerator/self_logs.go

```diff
@@ -113,14 +97,84 @@ func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Compon
 			},
 		},
 	}.Components(ctx, fluentBitSelfLogsTag, "self-logs-severity")...)
-	out = append(out, fluentbit.TranslationComponents(fluentBitSelfLogsTag, "severity", severityKey, true,
-		[]struct{ SrcVal, DestVal string }{
-			{"debug", "DEBUG"},
-			{"error", "ERROR"},
-			{"info", "INFO"},
-			{"warn", "WARNING"},
-		})...,
-	)
+	out = append(out, LoggingProcessorModifyFields{
+		Fields: map[string]*ModifyField{
+			"severity": {
+				MoveFrom: "jsonPayload.severity",
+				MapValues: map[string]string{
+					"error": "ERROR",
+					"warn":  "WARNING",
+					"info":  "INFO",
+					"debug": "DEBUG",
+				},
+				MapValuesExclusive: true,
+			},
+		},
+	}.Components(ctx, fluentBitSelfLogsTag, "mapseverityvalues")...)
+	return out
+}
+
+type healthLogsMatch struct {
+	key     string
+	match   string
+	message string
+	code    string
+}
+
+var healthLogsMatchList = []healthLogsMatch{
+	{
+		key:     "message",
+		match:   `\[error\]\s\[lib\]\sbackend\sfailed`,
+		message: "LogggingPipelineStoppedWorking-PleaseVerifyYourOpsAgentConfig",
+		code:    "LogPipelineErr",
+	},
+	{
+		key:     "message",
+		match:   `\[error\]\s\[parser\]\scannot\sparse`,
+		message: "IncorrectParsingConfiguration-PleaseVerifyYourOpsAgentConfig",
+		code:    "LogParseErr",
+	},
+}
+
+func generateSampleSelfLogsComponents(ctx context.Context) []fluentbit.Component {
+	out := make([]fluentbit.Component, 0)
+
+	// This filter will throttle all `ops-agent-health` logs by limiting the processing
+	// and ingesting to 30 logs every hour. 
+	out = append(out, fluentbit.Component{
```

Comment:
Just a reminder that we discussed this and decided to remove the throttle.

Reply:
Removed.